mapper/orm/src/main/java/org/hibernate/search/mapper/orm/reporting/impl/HibernateOrmMapperHints.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.reporting.impl;
+
+import org.hibernate.search.mapper.orm.cfg.HibernateOrmMapperSettings;
+import org.hibernate.search.mapper.pojo.reporting.spi.MapperHints;
+import org.hibernate.search.util.common.logging.impl.MessageConstants;
+
+import org.jboss.logging.Messages;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageBundle;
+
+@MessageBundle(projectCode = MessageConstants.PROJECT_CODE)
+public interface HibernateOrmMapperHints extends MapperHints {
+
+	HibernateOrmMapperHints INSTANCE = Messages.getBundle( HibernateOrmMapperHints.class );
+
+	@Message("Use \"" + HibernateOrmMapperSettings.MAPPING_CONFIGURER
+			+ "\" to supply a mapping configurer (org.hibernate.search.mapper.orm.mapping.HibernateOrmSearchMappingConfigurer)"
+			+ " that disables discovery of Jandex Indexes from added types,"
+			+ " or disables discovery of annotated types from root mapping annotations (e.g. @ProjectionConstructor)"
+			+ " if these are not used by the application."
+			+ " Configuration property \""
+			+ HibernateOrmMapperSettings.MAPPING_DISCOVER_ANNOTATED_TYPES_FROM_ROOT_MAPPING_ANNOTATIONS
+			+ "\" can be used instead of the configurer to disable discovery of annotated types from root mapping annotations."
+			+ " Alternatively, prebuild the Jandex indexes for these types and supply them through the configurer using context.annotationMapping().addJandexIndex(..).")
```

Comment:
Ok this is definitely more information, but not really clearer as it's quite complicated :x maybe let's talk about it on Zulip.